src/cassiopeia/critique/compare.py

```diff
@@ -8,110 +8,285 @@
+    tree2: CassiopeiaTree | str | nx.DiGraph,
+    tdata: TreeData | None = None,
     number_of_trials: int = 1000,
     min_triplets_at_depth: int = 1,
 ) -> tuple[dict[int, float], dict[int, float], dict[int, float], dict[int, float]]:
-    """Calculate the triplets correct accuracy between two trees.
+    """Calculate triplets-correct accuracy between two trees.
 
-    Takes in two newick strings and computes the proportion of triplets in the
-    tree (defined as a set of three leaves) that are the same across the two
-    trees. This procedure samples the same number of triplets at every depth
-    such as to reduce the amount of bias of sampling triplets randomly.
+    Accepts inputs as `CassiopeiaTree`, `networkx.DiGraph` (rooted, edges parent→child),
+    or string keys referencing trees stored in `tdata.obst`. Computes the proportion of
+    triplets (sets of three leaves) that agree between the two trees. Sampling is
+    stratified by depth (same number of triplets per depth) to reduce bias from
+    purely random triplet sampling. Unifurcations are collapsed before evaluation.
+    Trees must have identical leaf sets.
 
     Args:
-        tree1: Input CassiopeiaTree
-        tree2: CassiopeiaTree to be compared to the first tree.
-        number_of_trials: Number of triplets to sample at each depth
-        min_triplets_at_depth: The minimum number of triplets needed with LCA
-            at a depth for that depth to be included
+        tree1: First tree (`CassiopeiaTree`, `nx.DiGraph`, or str key into `tdata.obst`).
+        tree2: Second tree of the same type as `tree1`.
+        tdata: Required only when `tree1` and `tree2` are string keys; must provide `.obst`.
+        number_of_trials: Number of triplets to sample at each depth.
+        min_triplets_at_depth: Minimum number of triplets with LCA at a depth for that
+            depth to be included.
 
     Returns
     -------
-        A tuple of four dictionaries storing triplet statistics at each depth:
+        A tuple of four dicts keyed by depth:
             all_triplets_correct: Total proportion of triplets correct.
             resolvable_triplets_correct: Proportion correct among resolvable triplets.
             unresolved_triplets_correct: Proportion correct among unresolvable triplets.
-            proportion_resolvable: Proportion of triplets that are resolvable at each depth.
+            proportion_unresolvable: Proportion of triplets that are unresolvable at each depth.
     """
-    # keep dictionary of triplets correct
-    all_triplets_correct = defaultdict(int)
-    unresolved_triplets_correct = defaultdict(int)
-    resolvable_triplets_correct = defaultdict(int)
-    proportion_unresolvable = defaultdict(int)
+    if type(tree1) is not type(tree2):
+        raise TypeError("tree1 and tree2 must be the same type. ")
+
+    if isinstance(tree1, CassiopeiaTree):
+        G1 = to_nx_tree(copy.deepcopy(tree1))
+        G2 = to_nx_tree(copy.deepcopy(tree2))
+
```

Comment:
Do we need a `to_nx_tree` function?

For CassiopeiaTree objects, you can get this with `tree.get_tree_topology()`. For `treedata`, the objects in the `obst` slot should be networkx objects, no?